sycl/test-e2e/AOT/cpu.cpp

```diff
@@ -7,6 +7,7 @@
 //===---------------------------------------------------------------------===//
 
 // REQUIRES: opencl-aot, cpu
+// REQUIRES: build-and-run-mode
```

Comment:
What is the reason behind this extra requirement?

If we have any limitations now as to what we can do in our tests, then they should be documented somewhere.
Also, will we still continue to have `build-and-run` configuration somewhere in our pre-commit? I'm just double-checking that we are not disabling those tests forever.

Reply:
Hey, so the extra requirements in this pr fell into these camps:
- The tests inside the `Adapter` folder were testing sycl-ls, the `RUN:` lines there don't use `%{run}` or `%{run-unfiltered-devices}` so they were being ran on the build system. We could wrap these lines in `%if run-mode` but i figured it would just be cleaner to mark them as requires `build-and-run`, since they have no lines that compile to begin with
- The majority of the other tests were AOT cpu tests that fail on the run-side. These fail because when compiling the AOT tests we target the host machine's cpu architecture, which may be different than that of the running machine. So to get these tests running on separate machines we would need to restructure them to use the `-Xsycl-target-backend --march=` and then on the running side make sure we are executing the appropriate binary based on the cpu's architecture.
- the tests in `OptionalKernelFeatures` and `SpecConstants` that were changed failed on the build side due to member functions not being found. I'm honestly not too sure why/what the fix should be for these.

Here's a run of the CI before adding the extra requirements: https://github.com/intel/llvm/actions/runs/12262397938

As for the comment about the tests being disabled forever, I am working on #16306 to avoid this. The idea there being that when in `run-only` mode if we have a test marked as `REQUIRES: build-and-run` we know we will not have the binary available from the `build-only` step, so we should fully run the test, including any compilation steps.

Should also point out Andrei's comment on that pr https://github.com/intel/llvm/pull/16306#discussion_r1881197021, Which I think explains the current plan in regards to this extra mark-up.

Reply:
@AlexeySachkov Hey, just wanted to check if your concerns with the documentation + accidentally disabling tests were addressed. With the documentation pr #16409, and the use of the fallback option.

With the regards to the fallback option for running tests that require `build-and-run`, I've confirmed that on the split run-only side we are now running the tests marked with `REQUIRES: build-and-run-mode` that are able to run on gen12. (For example the `KernelAndProgram/test_cache_jit_aot.cpp` test which is marked this way ran, and passed there)